src/libs/ReportUtils.ts

```diff
@@ -4208,6 +4223,38 @@ function buildOptimisticApprovedReportAction(amount: number, currency: string, e
     };
 }
 
+/**
+ * Builds an optimistic APPROVED report action with a randomly generated reportActionID.
+ */
+function buildOptimisticUnapprovedReportAction(amount: number, currency: string, expenseReportID: string): OptimisticUnapprovedReportAction {
+    const originalMessage = {
```

Comment:
Personally I don't see the need to extract this if we're not using `originalMessage` multiple times below

Reply:
agree